src/rust/src/metrics_ffi.rs

```diff
@@ -171,7 +173,7 @@ pub extern "C" fn metrics_increment_counter(key: *mut FfiKey, value: u64) {
 pub extern "C" fn metrics_static_update_gauge(callsite: *const FfiCallsite, value: c_double) {
     if let Some(recorder) = try_recorder() {
         let callsite = unsafe { callsite.as_ref().unwrap() };
-        recorder.update_gauge(&callsite.key, GaugeValue::Absolute(value));
+        recorder.register_gauge(&callsite.key).set(value as f64);
```

Comment:
The reason for the `as f64` is that `c_double` is not specified to be equivalent to `f64` (it may be a different floating-point type, which we would just want to convert in that case).